python/lsst/dax/obscore/obscore_exporter.py

```diff
@@ -312,6 +320,113 @@ def to_csv(self, output: str) -> None:
+
+        tables = schema.tables
+        if len(tables) != 1:
+            raise RuntimeError("More than one table defined in ObsCore schema")
+        obscore_columns = {column.name: column for column in tables[0].columns}
+
+        votable = astropy.io.votable.tree.VOTableFile()
+        resource = astropy.io.votable.tree.Resource()
+        votable.resources.append(resource)
+
+        fields = []
+        for arrow_field in self.schema:
+            if arrow_field.name in obscore_columns:
+                ffield = obscore_columns[arrow_field.name]
+                votable_datatype = FelisType.felis_type(ffield.datatype.value).votable_name
+                field = astropy.io.votable.tree.Field(
+                    votable,
+                    name=ffield.name,
+                    datatype=votable_datatype,
+                    arraysize=ffield.votable_arraysize,
+                    unit=ffield.ivoa_unit,
+                    ucd=ffield.ivoa_ucd,
+                )
+                fields.append(field)
+            elif arrow_field.name == "em_filter_name":
+                # Non-standard but part of internal standard schema.
+                field = astropy.io.votable.tree.Field(
+                    votable,
+                    name="em_filter_name",
+                    datatype="char",
+                    arraysize="*",
+                )
+                fields.append(field)
+            else:
+                # This must be a non-standard field. Attempt to add it.
+                kwargs = {}
+                datatype = ""
+                if arrow_field.type.equals(pyarrow.int64()):
+                    datatype = "int"
+                elif arrow_field.type.equals(pyarrow.string()):
+                    datatype = "char"
+                    kwargs["arraysize"] = "*"
+                elif arrow_field.type.equals(pyarrow.float64()):
+                    datatype = "double"
+                else:
+                    raise RuntimeError(f"Could not handle unrecognized column {arrow_field.name}")
+                field = astropy.io.votable.tree.Field(
+                    votable,
+                    name=arrow_field.name,
+                    datatype=datatype,
+                    **kwargs,
+                )
+                fields.append(field)
+
+        table0 = astropy.io.votable.tree.TableElement(votable)
+        resource.tables.append(table0)
+        table0.fields.extend(fields)
+
+        def is_none(v: Any) -> bool:
+            return v is None
+
+        chunks = []
+        n_rows = 0
+        for record_batch in self._make_record_batches(self.config.batch_size):
+            pydict = record_batch.to_pydict()
+            columns = []
+            for label, column in pydict.items():
+                # Need to mask out any None values.
+                mask = [is_none(v) for v in column]
+                mc = astropy.table.MaskedColumn(column, name=label, mask=mask)
```

Comment:
I do not know. Are you worried about memory usage of the extra mask?

Reply:
Well, memory is one concern, but also the presence of mask probably means that people have to use it which probably complicates things on client side? Again, I have no idea how it works.

Reply:
The mask is required for the stacking to work since some batches have None but others don't. I also think that the mask is needed for VOTable to serialize properly with the None in -- you get the string "None" instead of an empty value.

Reply:
Rewritten to use `arrow_to_numpy`.